### DISCUSSION TITLE
Cross-compile to mobile targets?

### DISCUSSION THREAD

**Discussion by flukejones** [slint-ui/slint#224] [Q&A] [answered]
Apologies, I'm not sure this is the best place to ask. But I have been looking for something in the Rust GUI space that could compete with Flutter on some level, and it looks like sixtyFPS may very well do that.

But I'm unsure if it can cross-compile to mobile targets or not. I'm not willing to invest much time in to trying unless I know for sure (I have a severe shortage of time).

**Comment by ogoffart** [comment, marked as answer]
Hi, this is a perfectly good place to ask.

If by mobile target you mean Android and iOS, this is something we plan on doing, but haven't invested any time on yet.
Fundamentally, this should work, since all our building blocks (the GL backend) should support mobile platform. At lease the `winit` rust crate we are using behind the scene.
That said, we haven't done any work yet to support these platform and it probably does not yet work out of the box yet. Build system issues, virtual keyboard integration, application lifetime, are just a some of the things we haven't looked at yet that might need some work.
But that's something we want to support eventually.

Tracking issues: https://github.com/sixtyfpsui/sixtyfps/issues/46 & https://github.com/sixtyfpsui/sixtyfps/issues/47

**Reply by flukejones** [reply]
Oh this is exciting. Looking forward to seeing how that turns out.

**Reply by Be-ing** [reply]
Very cool. Will you have some examples of convergent applications along the lines of Kirigami or libadwaita?
